protostar/starknet/abi.py

```diff
@@ -28,14 +32,29 @@ def get_function_parameters(contract_abi: AbiType, name: str) -> Dict[str, Cairo
 
 
 def find_abi_item(contract_abi: AbiType, name: str) -> Dict:
-    for item in contract_abi:
-        if item["name"] == name:
-            return item
-    raise AbiItemNotFoundException(f"Couldn't find '{name}' ABI")
+    try:
+        for item in contract_abi:
+            if item["name"] == name:
+                return item
+        raise AbiItemNotFoundException(f"Couldn't find '{name}' ABI")
+    except TypeError as ex:
+        raise AbiItemNotFoundException(str(ex)) from ex
 
 
 def has_abi_item(contract_abi: AbiType, name: str) -> bool:
     for item in contract_abi:
         if item["name"] == name:
             return True
     return False
+
+
+def load_abi(abi_path: Path) -> AbiType:
+    assert abi_path.suffix == ".json"
+    potential_abi = json.loads(abi_path.read_text())
+    panic_if_invalid_abi(potential_abi)
+    return potential_abi
+
+
+def panic_if_invalid_abi(abi: Any):
+    if not isinstance(abi, list):
+        raise ProtostarException("Invalid ABI.")
```

Comment:
Nit: Why this is a separate method? It's a single check used only in this one very place. I don't see why we should exclude every single check to a separate function. That would generate an unmaintainable code.

Reply:
It's not about separating checks. It's more about operating on one level of abstraction. ABI validation is quite a separate concern compared to loading ABI.

https://www.fabrizioduroni.it/2019/07/28/clean-code-functions/